File: day04-数组-元素移出/283.移除0/283.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int slow = 0, fast = 0, n = nums.size();
        while (fast < n)
        {
            if (nums[fast])
            {
                swap(nums[slow], nums[fast]);
                slow++;
            }
            fast++;
        }
    }
};

File: day04-数组-元素移出/283.移除0/283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let slow = 0,
    fast = 0,
    n = nums.size();
  while (fast < n) {
    if (nums[fast]) {
      swap(nums[slow], nums[fast]);
      slow++;
    }
    fast++;
  }
};
